plugins/goodix-fp/fu-goodixfp-common.c

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (C) 2016-2018 Richard Hughes <richard@hughsie.com>
+ * Copyright (C) 2020 boger wang <boger@goodix.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-goodixfp-common.h"
+
+/*
+ *  Crc functions
+ */
+
+#define WIDTH (8 * sizeof (guint32))
+#define FINAL_XOR_VALUE 0xFFFFFFFF
+#define REFLECT_DATA(X)         ((guint8) reflect ((X), 8))
+#define REFLECT_REMAINDER(X)    ((guint32) reflect ((X), WIDTH))
+
+
+guint8
+gx_proto_crc8_calc (guint8 *lubp_date, guint32 lui_len)
```

Comment:
we use `buf` and `bufsz` in other parts of the code. I think `buf` can be `const` here too.

Reply:
i think we could write common code for crc8 method,  because plugin vli also used.